src/main/scala/mesosphere/marathon/core/deployment/impl/StartingBehavior.scala

```diff
@@ -52,9 +52,7 @@ trait StartingBehavior extends ReadinessBehavior with StrictLogging { this: Acto
       launchQueue.add(runSpec, 1).pipeTo(self)
 
     case Sync => async {
-      val instances = await(instanceTracker.specInstances(runSpec.id))
-      val actualSize = instances.count { i => i.isActive || i.isReserved || i.isScheduled || i.isProvisioned }
-      val instancesToStartNow = Math.max(scaleTo - actualSize, 0)
+      val instancesToStartNow = await(instanceTracker.instancesToScheduleCount(runSpec.id, scaleTo))
```

Comment:
Hm, what is the benefit of moving it into the instance tracker? I always thought of the tracker as a simple CRUD interface.

Reply:
this is actually a step toward that crud, because it will let us to have the algorithm calculating how much to schedule in here ... so later we will just move to `schedule(appId, targetCount)`